lib/AccessoryManager.js

```diff
@@ -500,6 +507,200 @@ class AccessoryManager {
+         this._makeZwaveNodeValueResolver(onCharacteristicZwaveNodeValueCriteria),
+         Boolean
+       ))
+       .on('set', this._makeCharacteristicSetterForZwaveNodeValue(
+         onCharacteristicName,
+         onCharacteristicZwaveNodeValueCriteria,
+         onCharacteristicValueSet
+       ))
+
+       const hslValue = value => {
+         const rValue = this._zwave.findNodeValue(this._accessoryConfig.zwaveNodeId, {
+           class_id: COMMAND_CLASS_SWITCH_MULTILEVEL, // TODO: is this always going to be COMMAND_CLASS_SWITCH_BINARY?
+           instance: 3, // TODO: is this always going to be 1?
+           index: 0 // TODO: is this always going to be 0?
+         })
+         const gValue = this._zwave.findNodeValue(this._accessoryConfig.zwaveNodeId, {
+           class_id: COMMAND_CLASS_SWITCH_MULTILEVEL, // TODO: is this always going to be COMMAND_CLASS_SWITCH_BINARY?
+           instance: 4, // TODO: is this always going to be 1?
+           index: 0 // TODO: is this always going to be 0?
+         })
+         const bValue = this._zwave.findNodeValue(this._accessoryConfig.zwaveNodeId, {
+           class_id: COMMAND_CLASS_SWITCH_MULTILEVEL, // TODO: is this always going to be COMMAND_CLASS_SWITCH_BINARY?
+           instance: 5, // TODO: is this always going to be 1?
+           index: 0 // TODO: is this always going to be 0?
+         })
+         const wValue = this._zwave.findNodeValue(this._accessoryConfig.zwaveNodeId, {
+           class_id: COMMAND_CLASS_SWITCH_MULTILEVEL, // TODO: is this always going to be COMMAND_CLASS_SWITCH_BINARY?
+           instance: 6, // TODO: is this always going to be 1?
+           index: 0 // TODO: is this always going to be 0?
+         })
+         let hsl = convertToHSL(rValue.value, gValue.value, bValue.value);
+
+         return hsl
+       }
+
+       const hueCharacteristicValueSet = value => {
+         const brightness = brightnessCharacteristic.value
+         const saturation = saturationCharacteristic.value
+         const hue = value
+         //this._log(`Brightness: ${brightness} Hue: ${hue} Saturation: ${saturation}`)
+         const rgb = HSVtoRGB(hue,saturation,brightness)
+         //this._log(`Red: ${rgb.r} Green: ${rgb.g} Blue: ${rgb.b} White: ${rgb.w} `)
+         const colorValue = RGBWtoHex(rgb.r,rgb.g,rgb.b,rgb.w)
+         return colorValue
+       }
+
+       const hueCharacteristicName = 'Hue'
+       const hueCharacteristicZwaveNodeValueCriteria = {
+         commandClass: COMMAND_CLASS_SWITCH_MULTILEVEL,
+         valueIndex: SWITCH_BINARY_INDEX_SWITCH
+       }
+       const colorCharacteristicZwaveNodeValueCriteria = {
+         commandClass: 51,
+         valueIndex: SWITCH_BINARY_INDEX_SWITCH
+       }
+       hueCharacteristic
+       .on('get', this._makeCharacteristicGetterForZwaveNodeValue(
+         hueCharacteristicName,
+         this._makeZwaveNodeValueResolver(hueCharacteristicZwaveNodeValueCriteria),
+         hslValue.h
```

Comment:
`hslValue` is a function but you haven't called it? `h` won't be the value you are expecting here?